packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -84,6 +93,82 @@ export class AbstractQueryInterfaceTypeScript {
     return schemaNames.flatMap((value: any) => (value.schema_name ? value.schema_name : value));
   }
 
+  /**
+   * Describe a table structure
+   *
+   * This method returns an array of hashes containing information about all attributes in the table.
+   *
+   * ```js
+   * {
+   *    name: {
+   *      type:         'VARCHAR(255)', // this will be 'CHARACTER VARYING' for pg!
+   *      allowNull:    true,
+   *      defaultValue: null
+   *    },
+   *    isBetaMember: {
+   *      type:         'TINYINT(1)', // this will be 'BOOLEAN' for pg!
+   *      allowNull:    false,
+   *      defaultValue: false
+   *    }
+   * }
+   * ```
+   *
+   * @param tableName
+   * @param options Query options
+   *
+   */
+  async describeTable(tableName: TableNameOrModel, options?: DescribeTableOptions) {
+    let table: TableNameOrModel = { tableName: '' };
+
+    if (typeof tableName === 'string') {
+      table = { tableName };
+    }
+
+    if (typeof tableName === 'object' && tableName !== null) {
+      table = tableName;
+    }
```

Comment:
you could call `extractTableDetails` for this

Reply:
Done